src/waveform/renderers/waveformrendererpreroll.cpp

```diff
@@ -40,20 +40,21 @@ void WaveformRendererPreroll::draw(QPainter* painter, QPaintEvent* event) {
     bool preRollVisible = firstDisplayedPosition < 0;
     bool postRollVisible = lastDisplayedPosition > 1;
     if (preRollVisible || postRollVisible) {
-        double playMarkerPosition = m_waveformRenderer->getPlayMarkerPosition();
-        double samplesPerPixel = m_waveformRenderer->getVisualSamplePerPixel();
-        double numberOfSamples = m_waveformRenderer->getLength() * samplesPerPixel;
-
-        int currentPosition = m_waveformRenderer->getPlayPosVSample();
-        int totalSamples = m_waveformRenderer->getTotalVSample();
-        //qDebug() << "currentPosition" << currentPosition
-        //         << "lastDisplayedPosition" << lastDisplayedPosition
-        //         << "samplesPerPixel" << samplesPerPixel
-        //         << "numberOfSamples" << numberOfSamples
-        //         << "totalSamples" << totalSamples
-        //         << "WaveformRendererPreroll::playMarkerPosition=" << playMarkerPosition;
-
-        const int polyLength = static_cast<int>(40.0 / samplesPerPixel);
+        const double playMarkerPositionFrac = m_waveformRenderer->getPlayMarkerPosition();
+        const double vSamplesPerPixel = m_waveformRenderer->getVisualSamplePerPixel();
+        const double numberOfVSamples = m_waveformRenderer->getLength() * vSamplesPerPixel;
+
+        int currentVSamplePosition = m_waveformRenderer->getPlayPosVSample();
+        int totalVSamples = m_waveformRenderer->getTotalVSample();
+        // qDebug() << "currentVSamplePosition" << currentVSamplePosition
+        //          << "lastDisplayedPosition" << lastDisplayedPosition
+        //          << "vSamplesPerPixel" << vSamplesPerPixel
+        //          << "numberOfVSamples" << numberOfVSamples
+        //          << "totalVSamples" << totalVSamples
+        //          << "WaveformRendererPreroll::playMarkerPosition=" << playMarkerPositionFrac;
+
+        const double polyLength = 40.0 / vSamplesPerPixel;
+        const double polyVSampleSize = (polyLength + 1) * vSamplesPerPixel;
```

Comment:
The expression `polyLength + 1` appears multiple times in the following code. Can we assign it to a constant with a meaningful name that also communicates the *plus one* rationale?

Reply:
Done. I'm not 100% sure but think the `+ 1` is used for the offset to accomodate the line width of the QPainter. I changed it to `painter->pen().lineWidthF()` to make this explicit.